src/DependencyInjection/CacheExtension.php

```diff
@@ -36,12 +38,35 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new Loader\YamlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
 
         // Make sure config values are in the parameters
-        foreach (['router', 'session', 'doctrine', 'logging'] as $section) {
+        foreach (['router', 'session', 'doctrine', 'logging', 'annotation', 'serializer', 'validation'] as $section) {
             if ($config[$section]['enabled']) {
                 $container->setParameter('cache.'.$section, $config[$section]);
             }
         }
 
+        if ($config['annotation']['enabled']) {
+            $this->verifyDoctrineBridgeExists('annotation');
+            $container->register('cache.service.annotation', DoctrineCacheBridge::class)
+                ->setFactory('Cache\CacheBundle\Factory\AnnotationFactory::get')
```

Comment:
It does not have to. It works as it is. 
I did not know you could do like you suggested, Your way is better. I'll change